cmake/Options/BuildOptions.cmake

```diff
@@ -39,7 +39,7 @@ option(CMAKE_EXPORT_COMPILE_COMMANDS "cmake compile commands" ON)
 set(TILEDB_INSTALL_LIBDIR "" CACHE STRING "If non-empty, install TileDB library to this directory instead of CMAKE_INSTALL_LIBDIR.")
 
 if (NOT TILEDB_VCPKG)
-  message(DEPRECATION "Disabling TILEDB_VCPKG is deprecated and will be removed in a future version.")
+  message(FATAL_ERROR "Disabling TILEDB_VCPKG is not supported.")
```

Comment:
For now I made this a fatal error. In the near future `TILEDB_VCPKG` will be completely removed, and passing `-DTILEDB_VCPKG=OFF` will result in the usual CMake warning about unused variables.

Reply:
OK.